base_view_inheritance_extension/models/ir_ui_view.py

```diff
@@ -0,0 +1,131 @@
+        return UnquoteObject("{}.{}".format(self, name))
+
+    def __repr__(self):
+        return self
+
+    def __call__(self, *args, **kwargs):
+        return UnquoteObject(
+            "%s(%s)"
+            % (
+                self,
+                ",".join(
+                    [
+                        UnquoteObject(a if not isinstance(a, str) else "'%s'" % a)
+                        for a in args
+                    ]
+                    + ["{}={}".format(UnquoteObject(k), v) for (k, v) in kwargs.items()]
+                ),
+            )
+        )
+
+
+class UnquoteEvalObjectContext(tools.misc.UnquoteEvalContext):
+    def __missing__(self, key):
+        return UnquoteObject(key)
+
+
+class IrUiView(models.Model):
+    _inherit = "ir.ui.view"
+
+    @api.model
+    def apply_inheritance_specs(self, source, specs_tree, inherit_id):
+        for specs, handled_by in self._iter_inheritance_specs(specs_tree):
+            source = handled_by(source, specs, inherit_id)
+        return source
+
+    @api.model
+    def _iter_inheritance_specs(self, spec):
+        if spec.tag == "data":
+            for child in spec:
+                for node, handler in self._iter_inheritance_specs(child):
+                    yield node, handler
+            return
+        if spec.get("position") == "attributes":
+            if all(not c.get("operation") for c in spec):
+                yield spec, self._get_inheritance_handler(spec)
+                return
+            for child in spec:
+                node = etree.Element(spec.tag, **spec.attrib)
+                node.insert(0, child)
+                yield node, self._get_inheritance_handler_attributes(child)
+            return
+        yield spec, self._get_inheritance_handler(spec)
+
+    @api.model
+    def _get_inheritance_handler(self, node):
+        handler = super(IrUiView, self).apply_inheritance_specs
+        if hasattr(self, "inheritance_handler_%s" % node.tag):
+            handler = getattr(self, "inheritance_handler_%s" % node.tag)
+        return handler
+
+    @api.model
+    def _get_inheritance_handler_attributes(self, node):
+        handler = super(IrUiView, self).apply_inheritance_specs
+        if hasattr(self, "inheritance_handler_attributes_%s" % node.get("operation")):
+            handler = getattr(
+                self, "inheritance_handler_attributes_%s" % node.get("operation")
+            )
+        return handler
+
+    @api.model
+    def inheritance_handler_attributes_python_dict(self, source, specs, inherit_id):
+        """Implement
+        <$node position="attributes">
+            <attribute name="$attribute" operation="python_dict" key="$key">
+                $keyvalue
+            </attribute>
+        </$node>"""
+        node = self.locate_node(source, specs)
+        for attribute_node in specs:
+            python_dict = safe_eval(
+                node.get(attribute_node.get("name")) or "{}",
+                UnquoteEvalObjectContext(),
```

Comment:
I noticed a problem while porting module `account_cost_center` to V13.

Consider the following code:

```xml
    <record id="view_move_form" model="ir.ui.view">
        <field name="model">account.move</field>
        <field name="inherit_id" ref="account.view_move_form" />
        <field name="arch" type="xml">
            <field name="ref" position="after">
                <field name="cost_center_id" />
            </field>
            <field name="invoice_line_ids" position="attributes">
                <attribute name="context" operation="python_dict" key="cost_center_id">cost_center_id</attribute>
            </field>
        </field>
    </record>
```

The above applies to line [account/views/account_move_views.xml#L665](https://github.com/odoo/odoo/blob/13.0/addons/account/views/account_move_views.xml#L665) so that the expected result should be:

`{ ... bla bla ... , 'default_currency_id': currency_id != company_currency_id and currency_id or False, 'cost_center_id': cost_center_id}`

Instead this is what I get:

`{ ... bla bla ... , 'default_currency_id': currency_id, 'cost_center_id': cost_center_id}`


I think it depends on the `UnquoteObject` implementation.